apps/src/javalab/JavalabEditor.jsx

```diff
@@ -23,31 +27,42 @@ const style = {
   },
   button: {
     marginLeft: 10
+  },
+  renameForm: {
+    marginBottom: 0,
+    display: 'flex',
+    alignItems: 'center'
   }
 };
 
 class JavalabEditor extends React.Component {
   static propTypes = {
     style: PropTypes.object,
     // populated by redux
-    editorText: PropTypes.string,
     setEditorText: PropTypes.func,
-    fileName: PropTypes.string,
-    setFileName: PropTypes.func
+    setFilename: PropTypes.func,
+    filename: PropTypes.string,
+    editorText: PropTypes.string
   };
 
   constructor(props) {
     super(props);
 
+    this.activateRenameFile = this.activateRenameFile.bind(this);
+    this.renameFileComplete = this.renameFileComplete.bind(this);
+    this.onSave = onSave.bind(this);
```

Comment:
if these 3 functions are arrow functions, you don't need to manually bind them -- they'll automatically be bound to `this`

Reply:
if you do that, you can also get rid of the constructor and set the initial component state in the root of the component

Reply:
the reason I bound them here was because of this part of the [style guide](https://github.com/airbnb/javascript/tree/master/react#methods) Jessie mentioned--apparently using arrow functions for event handlers can slow things down.

Reply:
TIL! thanks for sharing that -- you can ignore that comment 😄 you can still set the initial state outside of the constructor, but that seems less necessary if we'll be keeping the constructor anyways

Reply:
do we have a preference for initializing state inside/outside the constructor? Didn't know before today you could initialize it outside :)

Reply:
i don't think we have an explicit preference. what you have aligns with what i've seen in our codebase -- set initial state in the component root if there isn't a constructor, otherwise set initial state inside the constructor